python/pyarrow/_dataset.pyx

```diff
@@ -426,66 +432,96 @@ cdef class FileFragment(Fragment):
         return FileFormat.wrap(self.file_fragment.format())
 
 
-cdef class ParquetFileFormatReaderOptions:
-    cdef:
-        CParquetFileFormatReaderOptions* options
-
-    def __init__(self, ParquetFileFormat fmt):
-        self.options = &fmt.parquet_format.reader_options
-
-    @property
-    def use_buffered_stream(self):
-        """Read files through buffered input streams rather than
-        loading entire row groups at once. This may be enabled to
-        reduce memory overhead. Disabled by default."""
-        return self.options.use_buffered_stream
+cdef class ParquetFileFormat(FileFormat):
+    """
+    Format and options for reading Parquet files.
 
-    @use_buffered_stream.setter
-    def use_buffered_stream(self, bint value):
-        self.options.use_buffered_stream = value
+    Parameters
+    ----------
+    use_buffered_stream : bool, default False
+        Read files through buffered input streams rather than loading entire
+        row groups at once. This may be enabled to reduce memory overhead.
+        Disabled by default.
+    buffer_size : int, default 8192
+        Size of buffered stream, if enabled. Default is 8KB.
+    dictionary_columns : list of string, default None
+        Names of columns which should be read as dictionaries.
+    """
 
-    @property
-    def buffer_size(self):
-        """Size of buffered stream, if enabled. Default is 8KB."""
-        return self.options.buffer_size
+    cdef:
+        CParquetFileFormat* parquet_format
 
-    @buffer_size.setter
-    def buffer_size(self, int value):
-        self.options.buffer_size = value
+    def __init__(self, bint use_buffered_stream=False,
+                 int buffer_size=8192, dictionary_columns=None):
```

Comment:
I would maybe have a default of None for buffer_size, which would mean the default? Because now this default is defined in C++, but also hardcoded here (so in case we would change that, we need to change it here as well)

Reply:
Updating. Note that we'll still need to update the docstring.

Reply:
I'd rather keep these default, it's more explicit and semantically meaningful. For example if I want to retrieve the read_options from the format:

```python
format = ParquetFormat(ParquetReadOptions(use_buffered_stream=True))
format.read_options.use_buffered_stream is True
format.read_options.buffer_size is None  # ???
```

Additionally I'd need a lot of none checking.

Reply:
If you keep it, can you add a comment where this constant is defined in the C++ code that, if updated, the value also needs to be updated in the python code?

Reply:
Good idea, adding.